service/topology_coordinator.cc

```diff
@@ -740,6 +755,56 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
         }
     }
 
+    // If a node crashes after initiating gossip and before joining group0, it becomes orphan and 
+    // remains in gossip. This backgound fiber periodically checks for such nodes and purges those 
+    // entries from gossiper by commiting the node as left in raft.
+    future<> gossiper_orphan_remover_fiber() {
+        rtlogger.debug("start gossiper orphan remover fiber");
+
+        while (!_as.abort_requested()) {
+            try {
+                auto guard = co_await start_operation();
+                std::vector<canonical_mutation> updates;
+
+                _gossiper.for_each_endpoint_state([&](const gms::inet_address& addr, const gms::endpoint_state& eps) -> void {
+                    auto current_timestamp =
+                            std::chrono::duration_cast<std::chrono::seconds>(std::chrono::high_resolution_clock::now().time_since_epoch()).count();
+                    auto generation = eps.get_heart_beat_state().get_generation().value();
+                    static constexpr int32_t timeout = 60;
+                    auto host_id = eps.get_host_id();
+                    if (current_timestamp - generation > timeout && !_topo_sm._topology.contains(raft::server_id{host_id.id})) {
+                        topology_mutation_builder builder(guard.write_timestamp());
+                        builder.with_node(raft::server_id{host_id.id})
+                                .set("datacenter", get_application_state(host_id, addr, eps.get_application_state_map(), gms::application_state::DC))
+                                .set("rack", get_application_state(host_id, addr, eps.get_application_state_map(), gms::application_state::RACK))
+                                .set("release_version", version::release())
+                                .set("num_tokens", _db.get_config().join_ring() ? _db.get_config().num_tokens() : 0)
+                                .set("tokens_string", _db.get_config().join_ring() ? _db.get_config().initial_token() : sstring())
+                                .set("shard_count", smp::count)
+                                .set("ignore_msb", _db.get_config().murmur3_partitioner_ignore_msb_bits())
+                                .set("request_id", utils::UUID_gen::get_time_UUID())
+                                .set("cleanup_status", cleanup_status::clean)
```

Comment:
Yeah, these fields are required ones!!

Lets take all the fields from topology coordinator only, since host id is already there to distinguish. fields like dc and rack cant be dummy(bad UX), so lets have all the fields corresponding to coordinator(at least those are real in that case)

I don't have strong opinion here, lemme know the same!!